### PR TITLE
Oral-B Bluetooth Toothbrush

### PR DESCRIPTION
**Pull request by DigiH**
Initial submission of Oral-B Bluetooth Toothbrush decoder

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/theengs/decoder/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/BLE/test_ble.cpp [hunk 1]
```diff
@@ -468,6 +478,16 @@ const char* test_mfgdata[][3] = {
...
+    {"Oral-B", "Oral-B", "dc0004710502360000000f0004"},
+    {"Oral-B", "Oral-B", "dc000471050332010301030a04"},
+    {"Oral-B", "Oral-B", "dc00047105013a000002010004"},
+    {"Oral-B", "Oral-B", "dc000471050432010e03032e04"},
+    {"Oral-B", "Oral-B", "dc000471057332010e05032e04"},
+    {"Oral-B", "Oral-B", "dc000471050332010e06032e04"},
+    {"Oral-B", "Oral-B", "dc000471050332010e07032e04"},
+    {"Oral-B", "Oral-B", "dc000471050332010e08032e04"},
+    {"Oral-B", "Oral-B", "dc000202060220000001010004"},
+    {"Oral-B", "Oral-B", "dc000202067320020f07080004"},
```
Comment (kabili207) [inline tests/BLE/test_ble.cpp:490]: And here too

File: tests/BLE/test_ble.cpp [hunk 2]
```diff
@@ -121,6 +121,16 @@ const char* expected_mfg[] = {
...
+    "{\"brand\":\"Oral-B\",\"model\":\"BT Toothbrush\",\"model_id\":\"OralB_BT\",\"type\":\"BODY\",\"state\":\"sleeping\",\"mode\":\"deep clean\",\"sector\":\"sector 3\",\"pressure\":50,\"time\":74}",
+    "{\"brand\":\"Oral-B\",\"model\":\"BT Toothbrush\",\"model_id\":\"OralB_BT\",\"type\":\"BODY\",\"state\":\"running\",\"mode\":\"tongue cleaning\",\"sector\":\"sector 3\",\"pressure\":50,\"time\":74}",
+    "{\"brand\":\"Oral-B\",\"model\":\"BT Toothbrush\",\"model_id\":\"OralB_BT\",\"type\":\"BODY\",\"state\":\"running\",\"mode\":\"turbo\",\"sector\":\"sector 3\",\"pressure\":50,\"time\":74}",
+    "{\"brand\":\"Oral-B\",\"model\":\"BT Toothbrush\",\"model_id\":\"OralB_BT\",\"type\":\"BODY\",\"state\":\"running\",\"sector\":\"sector 3\",\"pressure\":50,\"time\":74}",
+    "{\"brand\":\"Oral-B\",\"model\":\"BT Toothbrush\",\"model_id\":\"OralB_BT\",\"type\":\"BODY\",\"state\":\"idle\",\"mode\":\"daily clean\",\"sector\":\"sector 1\",\"pressure\":32,\"time\":0}",
+    "{\"brand\":\"Oral-B\",\"model\":\"BT Toothbrush\",\"model_id\":\"OralB_BT\",\"type\":\"BODY\",\"state\":\"sleeping\",\"mode\":\"turbo\",\"sector\":\"sector 8\",\"pressure\":32,\"time\":135}",
```
Comment (kabili207) [inline tests/BLE/test_ble.cpp:133]: Same for the `model_id`s here

File: src/devices/OralB_json.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+/*R""""(
+{
+   "brand":"Oral-B",
+   "model":"BT Toothbrush",
+   "model_id":"OralB_BT",
```
Comment (kabili207) [inline src/devices/OralB_json.h:6]: The `model_id` would need to be all caps as well

File: src/decoder.h [hunk 4]
```diff
@@ -118,6 +118,7 @@ class TheengsDecoder {
     ABN07,
     ABTEMP,
     AMPHIRO,
+    OralB_BT,
```
Comment (kabili207) [inline src/decoder.h:121]: This should probably be in all caps for consistency.

**Comment by kabili207** [comment]
Ah, seems I wasn't quick enough to get my comments in before it was merged

**Comment by DigiH** [comment]
I suppose so ;) that's what you get digging out ancient decoder trials and focusing more on the functionality than consistency.

Thanks for pointing it out! :) will sort it in a new commit

File: tests/BLE/test_ble.cpp (hunk 1 above)
Comment (DigiH) [reply tests/BLE/test_ble.cpp:490]: These are fine though as they only specify the test name and device braodcast name, with the latter not relevant for this decoder.

**Comment by kabili207** [comment]
Hahaha, yeah I've done that myself a few times.

I've got one of these brushes myself, so it's nice to see someone adding a decoder for it

**Comment by DigiH** [comment]
The decoder might still require some refinements in the future, especially with so many different Oral-B model around.

If you find any issues or suggestions for additions with your model in the future any comments will be appreciated.

**Pull request merged**